### PR TITLE
Initialize reimbursement page

### PR DESCRIPTION
**Pull request by drazik**
This adds several things around reimbursements.

First, it cleans the health transactions with reimbursements in fixtures. They are all matched apart 3 of them.

Second, it introduces the `Reimbursements` account type. For now, one such account has been added to the fixtures, but in the future it will be a virtual account/group generated automatically. From this account type, a virtual group is created and is visible at the last position in the « balances » page. It is also unchecked by default.

Last, when we click on the `Reimbursements` account, we are not redirected to the `/transactions` route, but to a new `/balances/reimbursements` route that is a child of the `balances` route. That's why the selected tab remains « balances ». For now, this new page renders a `TransactionsPage` with just a `BackButton` to return to the « balances » page.

You can see it live on https://testbanksdemoreimbursements-banks.cozy.works

### CODE REVIEW

File: src/ducks/reimbursements/ReimbursementsPage.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { TransactionsPage } from 'ducks/transactions'
+
+class ReimbursementsPage extends React.PureComponent {
```
Comment (kosssi) [inline src/ducks/reimbursements/ReimbursementsPage.jsx]: J'ai l'impression que tu peux supprimer ce fichier et faire comme [ici](https://github.com/cozy/cozy-banks/pull/931/files#diff-74afc5ac3c5c28eb8d1c3d5227c062c9R42).

Comment (drazik) [reply src/ducks/reimbursements/ReimbursementsPage.jsx]: I refactored the code to export a `TransactionsPageWithBackButton` that I use in both placed :+1:

File: src/components/AppRoute.jsx [hunk 2]
```diff
@@ -29,13 +29,19 @@ const AppRoute = (
...
+            component={TransactionsPageWithBackButton}
+          />
+        </Route>
+        <Route path="transactions" component={TransactionsPageWithBackButton} />
         <Route path="categories">
           <IndexRoute component={CategoriesPage} />
           <Route
             path=":categoryName/:subcategoryName"
-            component={TransactionsPage}
+            component={props => <TransactionsPage {...props} showBackButton />}
```
Comment (kosssi) [inline src/components/AppRoute.jsx]: I think it's `TransactionsPageWithBackButton` to.

File: src/components/AppRoute.jsx [hunk 3]
```diff
@@ -29,13 +29,19 @@ const AppRoute = (
...
-        <Route path="balances" component={Balance} />
-        <Route path="transactions" component={TransactionsPage} />
+        <Route path="balances">
+          <IndexRoute component={Balance} />
+          <Route
+            path="reimbursements"
+            component={TransactionsPageWithBackButton}
+          />
+        </Route>
+        <Route path="transactions" component={TransactionsPageWithBackButton} />
```
Comment (kosssi) [inline src/components/AppRoute.jsx]: why you add backbutton here ?

File: src/ducks/transactions/TransactionRow.jsx [hunk 4]
```diff
@@ -92,8 +93,8 @@ class _RowDesktop extends React.PureComponent {
                 {!filteringOnAccount && (
                   <Caption className={styles['bnk-op-desc-caption']}>
                     {getAccountLabel(account)}
-                    {' - '}
-                    {getAccountInstitutionLabel(account)}
+                    {accountInstitutionLabel && ' - '}
```
Comment (kosssi) [inline src/ducks/transactions/TransactionRow.jsx]:
```suggestion
                    {accountInstitutionLabel && ` - ${accountInstitutionLabel}`}
```

File: src/components/AppRoute.jsx (hunk 3 above)
Comment (drazik) [reply src/components/AppRoute.jsx]: Mistake, I fixed it.

File: src/components/AppRoute.jsx (hunk 2 above)
Comment (drazik) [reply src/components/AppRoute.jsx]: Yes, thanks.

File: src/ducks/balance/components/AccountRow.jsx [hunk 5]
```diff
@@ -91,8 +91,7 @@ class AccountRow extends React.PureComponent {
               styles['AccountRow__column--secondary']
             )}
           >
-            N°
-            {account.number}
+            {account.number && `N°${account.number}`}
```
Comment (kosssi) [inline src/ducks/balance/components/AccountRow.jsx]: You can check `account.number` [here](https://github.com/cozy/cozy-banks/pull/931/files#diff-1eb84ac8f90830838bf8662f521b2539R87) to not print empty div no ?

File: src/ducks/balance/components/GroupPanel.jsx [hunk 6]
```diff
@@ -72,8 +72,16 @@ class GroupPanel extends React.PureComponent {
   }
 
   goToTransactionsFilteredByDoc = () => {
-    this.props.filterByDoc(this.props.group)
-    this.props.router.push('/transactions')
+    const { group } = this.props
```
Comment (kosssi) [inline src/ducks/balance/components/GroupPanel.jsx]: maybe you can import `filterByDoc` and `router` directly here

File: src/ducks/balance/components/AccountsList.jsx [hunk 7]
```diff
@@ -17,7 +17,13 @@ class AccountsList extends React.PureComponent {
...
   goToTransactionsFilteredByDoc = account => () => {
     this.props.filterByDoc(account)
-    this.props.router.push('/transactions')
+
+    const isReimbursementsType = getAccountType(account) === 'Reimbursements'
+    const route = isReimbursementsType
+      ? '/balances/reimbursements'
+      : '/transactions'
+
+    this.props.router.push(route)
```
Comment (kosssi) [inline src/ducks/balance/components/AccountsList.jsx]: you can export `filterByDoc` and `router` from `this.props` on the head of this function

**Pull request merged**